tests/test_pulse.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import pytest
+from inline_snapshot import snapshot
+
+from promdens.promdens import ENVELOPE_TYPES, LaserPulse
+
+def test_invalid_envelope_type(make_pulse):
+    with pytest.raises(ValueError):
```

Comment:
Here we test that the code raises ValueError if we pass the wrong envelope type.